User service Api create & get permissions

Preethi-Sl opened the pull request:
User Service - Create and Get API permissions

Preethi-Sl committed:
User service Api create & get permissions

infa-rvarahag commented on src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java:
```diff
@@ -0,0 +1,145 @@
+        permissionName = "permname" + RandomStringUtils.randomAlphabetic(10).toLowerCase();
+    }
+
+    @Test(description = "Validates the creation of permissions")
```
description : Testing creation of permission with valid data

infa-rvarahag commented on src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java:
```diff
@@ -0,0 +1,145 @@
+        Response response = createPermission(permissionMap);
+        logger.info("Response Code: {}, Response Body: {}", response.getStatusCode(), response.getBody().asString());
+
+        if (response.getStatusCode() == 422 || response.getStatusCode() == 400) {
```
remove if statmes 
Just put this code in if else state statement place.

assertEquals(response.getStatusCode(), 201, "Permission creation failed with" + response.getStatusCode());
createdPermissionID = String.valueOf(response.jsonPath().getInt("result.Id"));
 Assert.assertNotNull(createdPermissionID, "Created Permission ID should not be null");

infa-rvarahag commented on src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java:
```diff
@@ -0,0 +1,145 @@
+        logger.info("Response Code: {}, \n Response Body: {}", response.getStatusCode(), response.getBody().asString());
+        assertEquals(response.getStatusCode(), 422, "Expected Bad Request for missing required fields - 422, but got " + response.getStatusCode());
+
+        String responseBody = response.getBody().asString();
```
this validation is not correct. because of  "Or  "  if anyone's condition true validation becomes true.
need to call Create permission multiple times and then need to validate for "message": ""  one after the other in the same method .

infa-rvarahag commented on src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java:
```diff
@@ -0,0 +1,145 @@
+        logger.info("Ended calling --------------- CreateUserRoles: Negative test completed with assertions on missing fields.");
+    }
+    @Test(description = "Fetches the list of available permissions already created")
+    public void getUserPermissions() {
```
"if you want to use globally move to the base class "MentorEDBaseTest" , else remove public .

infa-rvarahag commented on src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java:
```diff
@@ -0,0 +1,145 @@
+        }
+    }
+
+    public Response createPermission(HashMap<String, Object> permissionMap) {
```
same comment here, if you want to use globally  move to the base class "MentorEDBaseTest"
else remove public

infa-rvarahag commented on src/test/java/org/shikshalokam/backend/TestCRUDUserPermissions.java:
```diff
@@ -0,0 +1,145 @@
+        logger.info("Ended calling ---- GET Permissions: with all the assertions");
+    }
+
+    private URI createURI(String endpoint) {
```
this move to the MentorBase class so that all projects can use the same .
public static URI createURI(String endpoint)

all the Test cases  can use like this  :
MentorBase.createURI("endpoint URL");

Preethi-Sl committed:
Update & delete API

-Review comment fixes
-Update api completion for User service update permission api
-Delete api completion for User service delete permission api

Preethi-Sl committed:
User service Update & delete permission apis

infa-rvarahag committed:
Merge branch 'main' into Preethi-UsrService-CreateAndGetPermission
